src/main/content/_drafts/2018-07-19-installing-features-from-maven-dependencies.adoc

```diff
@@ -0,0 +1,183 @@
+
+== The runtime kernel
+
+If you're using Maven, here are the coordinates for the runtime kernel:
+[source,xml]
+----
+<assemblyArtifact>
+  <groupId>io.openliberty</groupId>
+  <artifactId>openliberty-kernel</artifactId>
+  <version>18.0.0.2</version>
+  <type>zip</type>
+</assemblyArtifact>
+----
+
+For Gradle:
+[source,groovy]
+----
+dependencies {
+   libertyRuntime 'io.openliberty:openliberty-kernel:18.0.0.2'
+}
+----
+
+== Dependencies
+
+The Open Liberty 18.0.0.2 features are available on https://search.maven.org/[The Central Repository] under the `io.openliberty.features` groupId.  When features are listed as dependencies, they will be installed by the Maven or Gradle plugin as well as provide Liberty API, SPI, and Java specification dependencies for compilation.  Features can also continue to be listed as before in the plugin configuration/attributes or in the `server.xml` file, in which case those features will also be retrieved from The Central Repository for installation.  The features that are retrieved will be stored in the local Maven or Gradle cache so that the next time you run the plugin, they won't need to be downloaded again.
+
+=== Maven
+The `features-bom` artifact provides the bill of materials (BOM) for each release version, and can be imported in the `dependencyManagement` section.  This allows each feature to be listed in the `dependencies` section without needing to specify the version.
+[source,xml]
+----
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>io.openliberty.features</groupId>
+                <artifactId>features-bom</artifactId>
+                <version>18.0.0.2</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.openliberty.features</groupId>
+            <artifactId>jaxrs-2.1</artifactId>
+            <type>esa</type>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+----
+
+=== Gradle
+Use the `libertyFeature` dependency configuration to install features. If you are using the `java` plugin, then the `libertyFeature` configuration extends from the `java` plugin's `compileOnly` configuration to provide Liberty API, SPI, and Java specification dependencies as well.
+[source,groovy]
+----
+dependencies {
+   libertyFeature 'io.openliberty.features:jaxrs-2.1:18.0.0.2'
```

Comment:
Is the version necessary here? I ask because the version needs to match between the kernel dependency and the jaxrs one and it means updating will require many changes in lock step. We don't need this for maven so we should be able to not need this for gradle. Perhaps we do now, but this is a possible area of improvement I think.

Reply:
Gradle does not natively have the ability to use boms.  It is possible with third party plugins, but that is up to the user and should be out of scope for this blog.